molecule/openstack/create.yml

```diff
@@ -3,6 +3,10 @@
   connection: local
   gather_facts: false
   vars:
+    # borrowing the cloud var from molecule_openstack_ci, so it's easier to
+    # switch to using that role after molecule is updated to let us use roles
+    # in create/destroy steps.
+    molecule_openstack_ci_cloud: default
```

Comment:
This is already merged, but it might be a good idea to pull from `{{ lookup('env', 'OS_CLOUD') }}` instead of hard-coding in "default" here. I feel like that one environment variable is not a terrible idea to keep around. It's pretty straightforward to update and change clouds.yaml and update that variable.

Reply:
I was on the fence about that, but opted to be explicit since molecule_openstack_ci was not pulling from the env: https://github.com/oasis-roles/molecule_openstack_ci/blob/master/defaults/main.yml#L17

Should we open an issue there to default to OS_CLOUD?

Reply:
I think it could be. What I REALLY want is the ability to specify `{{ lookup('env', 'OS_CLOUD', default='default') }}` or something equivalent. Because, looking through the lookup/env.py file, you'll see that they're calling `os.getenv(var, '')`. Why they don't allow us to specify our own default is beyond me. But this makes it a multi-step process if we want to detect an empty OS_CLOUD variable. We'd have to do either multiple levels of task with `set_fact` or something really ridiculous like

```yaml
molecule_openstack_ci_cloud: |
  {{ ( lookup('env', 'OS_CLOUD') == '' ) |
     ternary( 'default', lookup('env', 'OS_CLOUD')) }}
```

But that's as ugly as sin. I'd love a better solution than this. Any ideas?

Reply:
Today I learned about the second positional arg to `default`:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#defaulting-undefined-variables

> If you want to use the default value when variables evaluate to false or an empty string you have to set the second parameter to true:

```yaml
molecule_openstack_ci_cloud: |
  {{ lookup('env', 'OS_CLOUD') | default('default', true) }}
```

Reply:
We could also go nuts and make the default cloud name itself configurable, defaulting to 'default', so the role could be adapted to (maybe?) anyone's preferred clouds.yml conventions.

Reply:
Woah... mind == blown on that second parameter to default. I always wondered if it was limited to whether the variable was defined or not, so now I know that it is.

I dunno about configuring the default. That sounds like premature optimization to me. And definitely overkill. If your convention is for something other than "default", then that's why we have the OS_CLOUD variable!

Reply:
I opened up an issue on the role itself: https://github.com/oasis-roles/molecule_openstack_ci/issues/7

In that issue, I did indeed make fun of myself for even suggesting configuring a default for the default. :)